// quria/contents/core/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Core = void 0;
const http_request_1 = require("../../adapters/http-request");
const utils_1 = require("../../adapters/utils");
class Core {
    url;
    headers;
    constructor(url, headers) {
        this.url = url;
        this.headers = headers;
    }
    /**
     * List of available localization cultures
     * @returns List of available localization cultures
     */
    GetAvailableLocales() {
        const requestURL = `${this.url}/GetAvailableLocales/`;
        return (0, http_request_1.request)(requestURL, true, "GET", this.headers);
    }
    /**
     * Get the common settings used by the Bungie.Net environment.
     * @returns The common settings used by the Bungie.Net environment.
     */
    GetCommonSettings() {
        const requestURL = `${this.url}/Settings/`;
        return (0, http_request_1.request)(requestURL, true, "GET", this.headers);
    }
    /**
     * Get the user-specific system overrides that should be respected alongside common systems.
     * @returns The user-specific system overrides that should be respected alongside common systems.
     */
    GetUserSystemOverrides() {
        const requestURL = `${this.url}/UserSystemOverrides/`;
        return (0, http_request_1.request)(requestURL, true, "GET", this.headers);
    }
    /**
     * Gets any active global alert for display in the forum banners, help pages, etc. Usually used for DOC alerts.
     * @returns Any active global alert for display in the forum banners, help pages, etc. Usually used for DOC alerts.
     */
    GetGlobalAlerts(queryString) {
        const requestURL = (0, utils_1.formatQueryStrings)(`${this.url}/GlobalAlerts/`, queryString);
        return (0, http_request_1.request)(requestURL, true, "GET", this.headers);
    }
}
exports.Core = Core;
