// quria/contents/trending/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Trending = void 0;
const http_request_1 = require("../../adapters/http-request");
class Trending {
    url;
    headers;
    constructor(url, headers) {
        this.url = url;
        this.headers = headers;
    }
    /**
     * Returns trending items for Bungie.net, collapsed into the first page of items per category. For pagination within a category, call GetTrendingCategory.
     * @returns Trending items for Bungie.net, collapsed into the first page of items per category. For pagination within a category, call GetTrendingCategory.
     */
    GetTrendingCategories() {
        const requestURL = `${this.url}/Trending/Categories/`;
        return (0, http_request_1.request)(requestURL, true, "GET", this.headers);
    }
    /**
     * Returns paginated lists of trending items for a category.
     * @param categoryId The ID of the category for whom you want additional results.
     * @param pageNumber The page # of results to return.
     * @returns Paginated lists of trending items for a category.
     */
    GetTrendingCategory(categoryId, pageNumber) {
        const requestURL = `${this.url}/Trending/Categories/${categoryId}/${pageNumber}/`;
        return (0, http_request_1.request)(requestURL, true, "GET", this.headers);
    }
    /**
     * Returns the detailed results for a specific trending entry.
     * @param identifier The identifier for the entity to be returned.
     * @param trendingEntryType The type of entity to be returned.
     * @returns The detailed results for a specific trending entry.
     */
    GetTrendingEntryDetail(identifier, trendingEntryType) {
        const requestURL = `${this.url}/Trending/Details/${trendingEntryType}/${identifier}/`;
        return (0, http_request_1.request)(requestURL, true, "GET", this.headers);
    }
}
exports.Trending = Trending;
